### PR TITLE
Various corrections to documentation 

### PR DESCRIPTION
**Pull request by jtclemm**
**Summary**

Updates to pages 673-948 in #4274 

**Author(s)**

Joel Clemmer (SNL)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

### CODE REVIEW

File: doc/src/Python_scatter.rst [hunk 1]
```diff
@@ -23,19 +23,19 @@ passed by all calling processors, to individual atoms, which may be
...
 
 Note that the data returned by the gather methods,
-e.g. gather_atoms("x"), is different from the data structure returned
-by extract_atom("x") in four ways.  (1) Gather_atoms() returns a
-vector which you index as x[i]; extract_atom() returns an array
-which you index as x[i][j].  (2) Gather_atoms() orders the atoms
-by atom ID while extract_atom() does not.  (3) Gather_atoms() returns
-a list of all atoms in the simulation; extract_atoms() returns just
-the atoms local to each processor.  (4) Finally, the gather_atoms()
+e.g. :code:`gather_atoms("x")`, is different from the data structure returned
```
Comment (akohlmey) [inline doc/src/Python_scatter.rst]:
```suggestion
e.g. :py:meth:`gather_atoms("x") <lammps.lammps.gather_atoms()>`, is different from the data structure returned
```
Using ":py:meth:" as shown here should insert a link to the documentation of the method.

**Comment by akohlmey** [comment]
@jtclemm I am updating the bibliography and came across this:
```
**(Palermo)** Palermo, Wolf, Clemmer, O'Connor, in preparation.
```
This should be updated or deleted.

**Comment by jtclemm** [comment]
> This should be updated or deleted.

Deleted

**Comment by jtclemm** [comment]
@akohlmey I found poor placement of the STL marble race figure:

image

I'm not too familiar with rst, any ideas how to fix it?

**Comment by akohlmey** [comment]
> I'm not too familiar with rst, any ideas how to fix it?

This cannot be fixed in ReStructured Text, it is a result of LaTeX's figure placement algorithm. It will change as the document is modified and text added or removed and thus the location of the figure's "anchor" changes.
I think we have to accept it and hope that most of the time it is ok. Check out the latest rendering of the Manual after my changes plus those from Steve, Germain, and Trung: https://ci.lammps.org/job/dev/job/pull_requests/job/build_docs/10358/

**Comment by akohlmey** [comment]
@jtclemm this looks good. Iike I did with the other doc pull requests, I am pulling this into PR #4275 first to resolve overlaps and combine it with all other changes. I found two minor issues (there must not be a horizontal line '----------' at the end of an .rst file, and there was a missing closing bracket in one reference.

**Pull request merged**